app/models/application_type.rb

```diff
@@ -183,7 +183,8 @@ def code_menu
       end
     end
 
-    def menu(scope = by_name)
+    def menu(scope = by_name, type: nil)
```

Comment:
The mixing of `type` and `name` doesn't seem ideal, but I don't think it would make it clearer to call the parameter `type`, because then it'll be too easy to think it's referring to application_type when it's actually referring to application_type.type. Maybe that attribute of application_type should have been called category or similar, but it's too late now and not worth fixing.

Reply:
Actually I've just realised I got it backwards, and the model has a `name` parameter not a `type`. WOuld making the parameter to this method be `name` instead simplify things?

```suggestion
    def menu(scope = by_name, name: nil)
```

Reply:
I think I tried initially but name might be reserved as it gives `"ApplicationType"` for this class method